content/blog/kubernetes-chart-v4/index.md

```diff
@@ -0,0 +1,1033 @@
+                        Map.entry("namespace", "default")
+                    ))
+                ))
+            ))
+            .build(), ComponentResourceOptions.builder()
+                .dependsOn(argocd)
+                .build());
+
+        ctx.export("redisPassword", redisPasswordResource.result());
+    }
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+resources:
+  ns:
+    type: kubernetes:core/v1:Namespace
+    properties:
+      metadata:
+        name: argocd
+
+  redis-password:
+    type: random:RandomPassword
+    properties:
+      length: 16
+
+  redis-secret:
+    type: kubernetes:core/v1:Secret
+    properties:
+      metadata:
+        name: argocd-redis
+        namespace: ${ns.metadata.name}
+      type: Opaque
+      stringData:
+        auth: ${redis-password.result}
+    options:
+      retainOnDelete: true
+
+  argocd:
+    type: kubernetes:helm.sh/v4:Chart
+    properties:
+      chart: argo-cd
+      version: "6.11.1"
+      namespace: ${ns.metadata.name}
+      repositoryOpts:
+        repo: https://argoproj.github.io/argo-helm
+      values:
+        fullnameOverride: ""
+
+  guestbook:
+    type: kubernetes:yaml/v2:ConfigGroup
+    properties:
+      objs:
+      - apiVersion: argoproj.io/v1alpha1
+        kind: Application
+        metadata:
+          name: guestbook
+          namespace: ${ns.metadata.name}
+        spec:
+          project: default
+          source:
+            repoURL: https://github.com/argoproj/argocd-example-apps.git
+            targetRevision: HEAD
+            path: guestbook
+          destination:
+            server: https://kubernetes.default.svc
+            namespace: default
+    options:
+      dependsOn:
+      - ${argocd}
+
+outputs:
+  redisPassword: ${redis-password.result}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+The program creates the `argocd` namespace, installs the ArgoCD server, and then creates an ArgoCD `Application` resource.
+Observe how the program installs and uses a Custom Resource Definition (CRD) successfully, and uses `dependsOn`
+to ensure that the CRD is installed before using it.
+
+The `argo-cd` chart normally makes use of a Helm hook to initialize a password for the redis server.
+Since the Chart v4 resource doesn't support Helm hooks, this program creates the password directly.
+
+## Conclusion
+
+Pulumi loves empowering developers to use the best tools for the job, and we recommend using Helm charts to install
+third-party Kubernetes applications. Pulumi complements Helm by handling the cloud resources that are often required,
+such as an IAM Role or cloud storage bucket. Such combinations make for great reusable componentry.
+
+This is part of a broad initiative to use Multi-Language Component (MLC) technology to offer a consistent experience
+across the Pulumi SDKs. See [pulumi-kubernetes#1971](https://github.com/pulumi/pulumi-kubernetes/issues/1971) for details.
```

Comment:
This feels a bit to in the weeds for the blog (referencing MLCs, pointing to a somewhat messy epic ticket). If we want to situate this change, we could point to the ConfigFile/ConfigGroup blog and maybe docs for KustomizeDirectory and CustomResourceDefinitions, which have also been updated?